### DISCUSSION TITLE
How runner is connected with Docker?

### DISCUSSION THREAD

**Discussion by GopikaV24** [actions/actions-runner-controller#1990] [Questions]
Hi @mumoshu & Team,

Previously, we had created docker based GitHub selfhosted runner using [this](https://paullockaby.com/posts/2021/11/setting-up-github-runners-on-kubernetes/) documentation.

Now, we have enabled autoscaler using actions-runner-controller respository.

What is the difference in the way of docker connectivity between these two runners?

**Comment by toast-gear** [comment]
> What is the difference in the way of docker connectivity between these two runners?

Could you expand on which two runners (perhaps include a link)? A custom runner basd on that blog and?

**Reply by GopikaV24** [reply]
Custom runner based on [this](https://paullockaby.com/posts/2021/11/setting-up-github-runners-on-kubernetes/) blog vs actions-runner-controller runner

**Reply by toast-gear** [reply]
https://github.com/actions-runner-controller/actions-runner-controller/tree/master/runner you can review our runners as it's all open source :D. There're various things that we implement that people typically don't which makes things more reliable e.g proper signal handling via dumb-init, support for actions/runners various flags etc etc. It's open source so take a look. The main thing though is ARC isn't a general purpose GitHub runner orchetrator, it's a specific implementation. We test and make changes based on our runners and so if you're using an entirely custom runner with ARC not built on top of ours we are not likely to put much energy into debugging any problems should you run into any (the maintainers all have jobs etc) and we may make breaking changes that break custom runner set-ups.
